features/admin.workflow-models.v1 /components/create/configuration-details-form.tsx

```diff
@@ -0,0 +1,153 @@
+                console.log("Initial Values",initialValues)
+                temporarySteps.current.approvalSteps = initialValues.approvalSteps;
+            }
+        }, [initialValues]);
+        
+
+        // Exposing triggerSubmit method
+        useImperativeHandle(ref, () => ({
+            triggerSubmit: () => {
+                const configurationData: ConfigurationsFormValuesInterface = {
+                    approvalSteps: temporarySteps.current.approvalSteps
+                }
+                onSubmit?.(configurationData);
+                // if (triggerFormSubmit.current) {
+                //     triggerFormSubmit.current();
+                // }
+            }
+        }));
+
+        const [steps, setSteps] = useState<MultiStepApprovalTemplate[]>([]);
+        const temporarySteps = useRef<ConfigurationsFormValuesInterface>({
+            approvalSteps: []
+        });
+
+        // Add new approval step
+        const addNewStep = () => {
+            const newStep: MultiStepApprovalTemplate = {
+                id: `step${Date.now()}`,
+                stepNumber: steps.length + 1,
+                roles: [],
+                users: []
+            };
+            setSteps(prevSteps => [...prevSteps, newStep]); // Use state only, avoid refs here
+        };
+
+        // Handle step deletion
+        const handleDelete = (stepId: string, index: number) => {
+            setSteps(prevSteps => {
+                // Create a new array with the deleted step removed and re-number the steps
+                const newSteps = prevSteps
+                    .filter((step: MultiStepApprovalTemplate) => step.id !== stepId)
+                    .map((step: MultiStepApprovalTemplate, index: number) => ({
+                        ...step,
+                        stepNumber: index + 1 // Reassign step numbers
+                    }));
+                return newSteps;
+            });
+        
+            temporarySteps.current.approvalSteps = temporarySteps.current.approvalSteps.filter((_, i) => i !== index);
+
+            console.log("Updated after delete: ", temporarySteps.current);
+        };
+
+        // Handle changes in users or roles for specific steps
+        const handleStepChange = useCallback(
+            (index: number, updatedEntities: UserBasicInterface[] | RolesInterface[], type: "users" | "roles") => {
+                temporarySteps.current.approvalSteps[index] = {
+                    ...temporarySteps.current.approvalSteps[index],
+                    [type]: type === "users" ?  updatedEntities.map(user => user.userName): updatedEntities.map(role => role.displayName)
+                }
+                console.log("Updated after adding new values: ", temporarySteps.current);
+            },
+            []
+        );
+        useEffect(() => {
+            console.log("Received initialValues:", initialValues);
+          }, [initialValues]);       
+        
+        return (
+            <div>
```

Comment:
```suggestion
            <>
```
use fragement instead of `<div/>` to group elements